### PR TITLE
EVG 20071 Listen for MergeGroup event

### PR DESCRIPTION
**Pull request by bsamek**
- EVG-20070 Add GitHub merge queue ADR
- Listen for MergeGroupEvent

### CODE REVIEW

**Comment by bsamek** [comment]
Note that the ADR is in a separate PR, https://github.com/evergreen-ci/evergreen/pull/6581, but I included it here for context, since it isn't merged yet. The same reviewer for that PR should review this one @ablack12 .

File: rest/route/github.go
Comment (malikchaya2) [inline rest/route/github.go:227]: Is this meant to be temporary? If yes, can we please add a ticket number or comment indicating what it's doing and a todo to remove it?

File: docs/decisions/2023-06-05_github_merge_queue.md
Comment (malikchaya2) [inline docs/decisions/2023-06-05_github_merge_queue.md:1]: Will this be removed before merging and merged in [6581](https://github.com/evergreen-ci/evergreen/pull/6581/files) instead?

**Review by malikchaya2 — CHANGES_REQUESTED** [review]

File: docs/decisions/2023-06-05_github_merge_queue.md
Comment (malikchaya2) [reply docs/decisions/2023-06-05_github_merge_queue.md]: thank you!

File: rest/route/github.go
Comment (malikchaya2) [reply rest/route/github.go:227]: In that case, I would suggest adding a comment similar to [this format](https://github.com/malikchaya2/evergreen/blob/46deb4bceb4c3d745bc6250a60d04867851c08ed/rest/route/project.go#L477-L479).

**Pull request merged**
